comp/snmpscan/impl/devicescan.go

```diff
@@ -54,3 +57,72 @@ func gatherPDUs(snmp *gosnmp.GoSNMP) ([]*gosnmp.SnmpPDU, error) {
 	}
 	return pdus, nil
 }
+
+func (s snmpScannerImpl) ScanDeviceAndSendData(connParams *snmpparse.SNMPConfig, logger log.Component, namespace string) error {
```

Comment:
You shouldn't pass Component as method parameters, they should be dependencies of the current component.

So if you wanted to add a new dependency you should register it in `type Requires struct {` from `snmpscan.go`.
Good luck here, the logger is already a dependency so you don't need to do anything and in this function you can just use `s.log`

Reply:
udpated in [1b31315](https://github.com/DataDog/datadog-agent/pull/33960/commits/1b313150791a71b2bf6df3365e5c5f370c5d3458)